bin/compile

```diff
@@ -27,8 +27,8 @@ unset GIT_DIR
 trap build_failed ERR
 
 # Load config vars into environment; start with defaults
-export NPM_CONFIG_PRODUCTION=true
-export NODE_MODULES_CACHE=true
+export NPM_CONFIG_PRODUCTION=${NPM_CONFIG_PRODUCTION:-true}
+export NODE_MODULES_CACHE=${NODE_MODULES_CACHE:-true}
 export_env_dir $env_dir
 
```

Comment:
The line that loads the env vars (`export_env_dir`) is below these exports, so user config would never be loaded.

Let's get to the bottom of why you're not seeing `NPM_CONFIG_PRODUCTION` change your app's behavior; if that's the case, then these tests aren't working:

https://github.com/heroku/heroku-buildpack-nodejs/blob/master/bin/test#L202-L255